src/main/java/com/notably/commons/core/path/RelativePath.java

```diff
@@ -1,53 +1,112 @@
 package com.notably.commons.core.path;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
 import com.notably.commons.core.path.exceptions.InvalidPathException;
 
 /**
  * TODO: Add Javadoc
  */
 public class RelativePath implements Path {
+    public static final String INVALID_RELATIVE_PATH = "Invalid relative path";
+    public static final String VALIDATION_REGEX = "((\\.|\\..|\\p{Alnum}+))+\\/?";
```

Comment:
There is a bug in this `VALIDATION_REGEX`! It won't correctly match the `"asdfkj/sdfljk/asdlfkj"` path, for example, as the `/` characters between components are not detected!

Instead, we should rectify this as `"(\\.|\\..|\\p{Alnum}+)(\\/(\\.|\\..|\\p{Alnum}+))*\\/?"`